lib/Sema/ConstraintSystem.cpp

```diff
@@ -959,23 +959,25 @@ void ConstraintSystem::openGeneric(
       continue;
     }
 
-    ArchetypeType *archetype = ArchetypeBuilder::mapTypeIntoContext(dc, gp)
-                                 ->castTo<ArchetypeType>();
-    auto typeVar = createTypeVariable(getConstraintLocator(
+    Type mappedType = ArchetypeBuilder::mapTypeIntoContext(dc, gp);
+    if (mappedType) {
```

Comment:
It would mean the archetype builder was not set up properly.

You could try to see what the offending 'gp' actually is, and what generic signature it comes from vs what generic signature the archetype builder was configured with.

Usually getting to the root cause of something leads to better code and fixes more crashes.